src/object_store/src/object/disk.rs

```diff
@@ -500,6 +518,48 @@ mod tests {
         assert!(!path.exists());
     }
 
+    #[tokio::test]
+    #[cfg_attr(madsim, ignore)] // TODO: remove this when madsim supports fs
+    async fn test_delete_objects() {
+        let test_dir = TempDir::new().unwrap();
+        let test_root_path = test_dir.path().to_str().unwrap();
+        let store = DiskObjectStore::new(test_root_path);
+        let mut payload = gen_test_payload();
+
+        let str_list = vec![
+            String::from("test1.obj"),
+            String::from("test2.obj"),
+            String::from("test3.obj"),
+        ];
+
+        let path_list = str_list
+            .iter()
+            .map(|str| {
+                let mut path = PathBuf::from(test_root_path);
+                path.push(str);
+                path
+            })
+            .collect_vec();
+
+        for i in 0..2 {
```

Comment:
we can use `path_list.len()` , so that if we add more path for test, we no need to change the code of iter logic.

Reply:
I pushed an update to that function. It now has two constants stating how many "real" and "fake" files there are. The rest of that function then works based on that without hard-coded numbers.